src/Parse/Helpers.hs

```diff
@@ -156,6 +156,19 @@ spaceySepBy1 sep parser =
       (value:) <$> spaceyPrefixBy sep parser
 
 
+spaceySepByTrailing1 :: IParser b -> IParser a -> IParser [a]
+spaceySepByTrailing1 sep parser =
+  do  value <- parser
+      choice
+        [ do  try (padded sep)
+              choice
+                [ (value:) <$> spaceySepByTrailing1 sep parser
+                , return [value]
+                ]
```

Comment:
The `choice` can be simplified with `option`,  something like `option [value] $ (value:) <$> spaceySepByTrailing1 sep parser`